packages/core/src/modules/credentials/protocol/v2/__tests__/v2.ldproof.connectionless-credentials.test.ts

```diff
@@ -0,0 +1,204 @@
+
+    aliceAgent = new Agent(aliceConfig.config, aliceConfig.agentDependencies)
+    aliceAgent.registerInboundTransport(new SubjectInboundTransport(aliceMessages))
+    aliceAgent.registerOutboundTransport(new SubjectOutboundTransport(subjectMap))
+    await aliceAgent.initialize()
+
+    await prepareForIssuance(faberAgent, ['name', 'age'])
+
+    faberReplay = new ReplaySubject<CredentialStateChangedEvent>()
+    aliceReplay = new ReplaySubject<CredentialStateChangedEvent>()
+
+    faberAgent.events
+      .observable<CredentialStateChangedEvent>(CredentialEventTypes.CredentialStateChanged)
+      .subscribe(faberReplay)
+    aliceAgent.events
+      .observable<CredentialStateChangedEvent>(CredentialEventTypes.CredentialStateChanged)
+      .subscribe(aliceReplay)
+    wallet = faberAgent.injectionContainer.resolve(IndyWallet)
+    const issuerDidInfo = await wallet.createDid({ seed })
+    const issuerKey = Key.fromPublicKeyBase58(issuerDidInfo.verkey, KeyType.Ed25519)
+    issuerDidKey = new DidKey(issuerKey)
+    verificationMethod = `${issuerDidKey.did}#${issuerDidKey.key.fingerprint}`
+    const inputDoc = {
+      '@context': [
+        'https://www.w3.org/2018/credentials/v1',
+        'https://w3id.org/citizenship/v1',
+        'https://w3id.org/security/bbs/v1',
+      ],
+      id: 'https://issuer.oidp.uscis.gov/credentials/83627465',
+      type: ['VerifiableCredential', 'PermanentResidentCard'],
+      issuer: issuerDidKey.did,
+      identifier: '83627465',
+      name: 'Permanent Resident Card',
+      description: 'Government of Example Permanent Resident Card.',
+      issuanceDate: '2019-12-03T12:19:52Z',
+      expirationDate: '2029-12-03T12:19:52Z',
+      credentialSubject: {
+        id: 'did:example:b34ca6cd37bbf23',
+        type: ['PermanentResident', 'Person'],
+        givenName: 'JOHN',
+        familyName: 'SMITH',
+        gender: 'Male',
+        image: 'data:image/png;base64,iVBORw0KGgokJggg==',
+        residentSince: '2015-01-01',
+        lprCategory: 'C09',
+        lprNumber: '999-999-999',
+        commuterClassification: 'C1',
+        birthCountry: 'Bahamas',
+        birthDate: '1958-07-17',
+      },
+    }
+
+    credential = JsonTransformer.fromJSON(inputDoc, W3cCredential)
+
+    signCredentialOptions = {
+      credential,
+      proofType: 'Ed25519Signature2018',
+      verificationMethod,
+    }
+  })
+
+  afterEach(async () => {
+    await faberAgent.shutdown()
+    await faberAgent.wallet.delete()
+    await aliceAgent.shutdown()
+    await aliceAgent.wallet.delete()
+  })
+
+  test('Faber starts with V2 W3C connection-less credential offer to Alice', async () => {
+    testLogger.test('Faber sends credential offer to Alice')
+
+    const offerOptions: OfferCredentialOptions = {
+      comment: 'V2 Out of Band offer (W3C)',
+      credentialFormats: {
+        jsonld: signCredentialOptions,
+      },
+      protocolVersion: CredentialProtocolVersion.V2,
+      connectionId: '',
```

Comment:
connectionId should not be an allowed value for `createOutOfBandOffer` and definitely shouldn't be allowed as an empty string (this should blow up!)

```suggestion
```